lib/http/request/body.rb

```diff
@@ -33,8 +33,10 @@ def each
         if @body.is_a?(String)
           yield @body
         elsif @body.respond_to?(:read)
-          chunk = String.new # rubocop:disable Style/EmptyLiteral
-          yield chunk while @body.read(BUFFER_SIZE, chunk)
+          loop do
+            data = @body.read(BUFFER_SIZE) or break
```

Comment:
@tarcieri I would also prefer that, but Ruby prints a warning when you attempt to use an assignment in conditionals (Ruby thinks that you wanted to use `==`). It happens even when warning aren't enabled, just when running `ruby -e 'nil if a = 1'`.